packages/flutter_markdown/lib/src/markdown.dart

```diff
@@ -0,0 +1,472 @@
+    };
+  }
+}
+
+class Markdown extends StatefulComponent {
+  Markdown({
+    this.data,
+    this.markdownStyle,
+    this.syntaxHighlighter
+  });
+
+  final String data;
+  final MarkdownStyle markdownStyle;
+  final SyntaxHighlighter syntaxHighlighter;
+
+  State<Markdown> createState() => new _MarkdownState();
+}
+
+class _MarkdownState extends State<Markdown> {
+
+  void initState() {
+    super.initState();
+
+    MarkdownStyle markdownStyle = config.markdownStyle;
+    if (markdownStyle == null)
+      markdownStyle = new MarkdownStyle.defaultFromTheme(Theme.of(context));
+
+    SyntaxHighlighter syntaxHighlighter = config.syntaxHighlighter;
+    if (syntaxHighlighter == null)
+      syntaxHighlighter = new DartSyntaxHighlighter();
+
+    _cachedBlocks = _blocksFromMarkup(config.data, markdownStyle, syntaxHighlighter);
+  }
+
+  List<_Block> _cachedBlocks;
+
+  Widget build(BuildContext context) {
+    List<Widget> blocks = <Widget>[];
+    for (_Block block in _cachedBlocks) {
+      blocks.add(block.build(context));
+    }
+
+    return new Column(
+      alignItems: FlexAlignItems.stretch,
+      children: blocks
+    );
+  }
+}
+
+List<_Block> _blocksFromMarkup(String data, MarkdownStyle markdownStyle, SyntaxHighlighter syntaxHighlighter) {
+  var lines = data.replaceAll('\r\n', '\n').split('\n');
+  md.Document document = new md.Document();
+
+  _Renderer renderer = new _Renderer();
+  return renderer.render(document.parseLines(lines), markdownStyle, syntaxHighlighter);
+}
+
+class _Renderer implements md.NodeVisitor {
+  List<_Block> render(List<md.Node> nodes, MarkdownStyle markdownStyle, SyntaxHighlighter syntaxHighlighter) {
+    assert(markdownStyle != null);
+
+    _blocks = <_Block>[];
+    _listIndents = <String>[];
+    _markdownStyle = markdownStyle;
+    _syntaxHighlighter = syntaxHighlighter;
+
+    for (final md.Node node in nodes) {
+      node.accept(this);
+    }
+
+    return _blocks;
+  }
+
+  List<_Block> _blocks;
+  List<String> _listIndents;
+  MarkdownStyle _markdownStyle;
+  SyntaxHighlighter _syntaxHighlighter;
+
+  void visitText(md.Text text) {
+    List<dynamic> top = _currentBlock.stack.last;
+
+    if (_currentBlock.tag == 'pre')
+      top.add(_syntaxHighlighter.format(text.text));
+    else
+      top.add(text.text);
+  }
+
+  bool visitElementBefore(md.Element element) {
+    if (_isListTag(element.tag))
+      _listIndents.add(element.tag);
+
+    if (_isBlockTag(element.tag)) {
+      List<_Block> blockList;
+      if (_currentBlock == null)
+        blockList = _blocks;
+      else
+        blockList = _currentBlock.subBlocks;
+
+      _Block newBlock = new _Block(element.tag, element.attributes, _markdownStyle, new List<String>.from(_listIndents), blockList.length);
+      blockList.add(newBlock);
+    } else {
+      TextStyle style = _markdownStyle._styles[element.tag];
+      if (style == null)
+        style = new TextStyle();
+
+      List<dynamic> styleElement = <dynamic>[style];
```

Comment:
I don't think we should use this old-style data structure for formatting RichText. Instead, we should use the current data structures for RictText (i.e., TextSpan).